spec/requests/requests_spec.rb

```diff
@@ -3,85 +3,243 @@
+      end
     end
 
-    describe 'without hints param' do
-      let(:params) { { request: {  title: 'Example Question', text: 'How do you do?' } } }
-      it { should_not raise_error }
-    end
+    context 'unauthorized' do
+      let(:user) { create(:user, organizations: [create(:organization)]) }
 
-    describe 'without contributors' do
-      it { should_not raise_error }
+      it 'renders not found ' do
+        subject.call
+        expect(response).to be_not_found
+      end
     end
 
-    context 'with image file(s)' do
-      let(:params) do
-        { request: { title: 'Message with files', text: 'Did you get this image?', files: [fixture_file_upload('profile_picture.jpg')] } }
+    context 'authenticated and authorized' do
+      let(:user) { create(:user, organizations: [organization]) }
+
+      before do
+        create(:request, title: 'I belong to the organization', organization: organization)
+        create(:request, title: 'I do too!', organization: organization)
+        create(:request, title: 'Planned for the future', schedule_send_for: 1.day.from_now, broadcasted_at: nil,
+                         organization: organization)
+        create(:request, title: 'Also planned, but from another organization', schedule_send_for: 1.hour.from_now, broadcasted_at: nil)
+        create(:request, title: 'From another organization', organization: create(:organization))
+        subject.call
       end
 
-      describe 'an image file' do
-        it 'redirects to requests#show' do
-          response = subject.call
-          request = Request.last
-          expect(response).to redirect_to organization_request_path(organization, request)
+      it 'should be successful' do
+        subject.call
+        expect(response).to be_successful
+      end
+
+      context 'no params' do
+        it 'displays only sent messages' do
+          expect(page).to have_content('I belong to the organization')
+          expect(page).to have_content('I do too!')
+          expect(page).not_to have_content('Planned for the future')
         end
 
-        it 'shows success notification' do
-          subject.call
-          expect(flash[:success]).not_to be_empty
+        it 'display only requests from the organization' do
+          expect(page).to have_content('Gestellt 2')
+          expect(page).to have_content('Geplant 1')
+          expect(page).to have_content('I belong to the organization')
+          expect(page).to have_content('I do too!')
+          expect(page).not_to have_content('From another organization')
         end
       end
 
-      describe 'multiple image files' do
-        let(:params) do
-          { request: { title: 'Message with files', text: 'Did you get this image?',
-                       files: [fixture_file_upload('profile_picture.jpg'), fixture_file_upload('example-image.png')] } }
+      context 'sent filter param' do
+        let(:params) { { filter: :sent } }
+
+        it 'displays only sent messages' do
```

Comment:
does it behave the same as without parameters? Maybe a shared?

Reply:
I don't think adding a shared for one duplication would be the way to go.